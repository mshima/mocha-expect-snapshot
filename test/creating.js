const expect = require('expect');
const crypto = require('crypto');
const { unlinkSync, rmSync = unlinkSync, readFileSync } = require('fs');
const { resolve } = require('path');

const createSnapshot = false;

describe('creting test', function () {
  const tempSnapshotFile = resolve('test', '__snapshots__', 'creating.js.snap.' + crypto.randomBytes(4).readUInt32LE(0));
  const snapshotFile = resolve('test', '__snapshots__', 'creating.js.snap');

  before(function () {
    this.snapshotStateOptions = {
      updateSnapshot: 'all',
    };
    if (!createSnapshot) {
      this.snapshotFile = tempSnapshotFile;
      this.addAfterSnapshotSave((summary) => {
        expect(readFileSync(tempSnapshotFile).toString()).toBe(readFileSync(snapshotFile).toString());
        rmSync(tempSnapshotFile);

        expect(summary).toEqual(
          expect.objectContaining({
            added: 3,
            removed: 0,
            updated: 0,
          })
        );
      });
    }
  });

  describe('toMachSnapshot()', function () {
    it('should match one snapshot', function () {
      expect({ foo: 'bar' }).toMatchSnapshot();
    });

    it('should match two snapshots', function () {
      expect({ foo: 'bar' }).toMatchSnapshot();
      expect({ foo2: 'bar' }).toMatchSnapshot();
    });
  });
});
