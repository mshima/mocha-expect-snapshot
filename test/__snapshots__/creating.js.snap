// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creting test toMachSnapshot() should match one snapshot 1`] = `
{
  "foo": "bar",
}
`;

exports[`creting test toMachSnapshot() should match two snapshots 1`] = `
{
  "foo": "bar",
}
`;

exports[`creting test toMachSnapshot() should match two snapshots 2`] = `
{
  "foo2": "bar",
}
`;
